### PR TITLE
feat: UQI Pagination Component implementation 

### LINKED ISSUE #9348: [Feature]: Pagination UQI component
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

New component to be used with offset and limit options.

### Why should this be worked on?

This should make using paginations across plugins have the same experience.

### PR DESCRIPTION
**Pull request by Irongade**
New pagination UQI Component

Fixes #9348

- New feature (non-breaking change which adds functionality).

Manual Tests was employed.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feat/pagination-uqi-component 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.14 **(0.01)** | 37.18 **(0.01)** | 33.96 **(-0.01)** | 55.64 **(0)**
 :sparkles: :new: | **app/client/src/components/formControls/PaginationControl.tsx** | **60** | **100** | **25** | **58.62**
 :red_circle: | app/client/src/utils/FormControlRegistry.tsx | 41.67 **(-0.55)** | 100 **(0)** | 6.25 **(-0.42)** | 41.67 **(-0.55)**</details>

### CODE REVIEW

File: app/client/src/components/formControls/PaginationControl.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  display: flex;
+  flex-direction: column;
+  margin-bottom: 10px;
+`;
+
+// using query dynamic input text for both so user can dynamically change these values.
+const valueFieldConfig: any = {
+  key: "value",
+  controlType: "QUERY_DYNAMIC_INPUT_TEXT",
+  placeholderText: "value",
```
Comment (rashmi-rai) [inline app/client/src/components/formControls/PaginationControl.tsx:27]: We should be getting labels, placeholderText etc from API. 
No need to hard code here

Comment (Irongade) [reply app/client/src/components/formControls/PaginationControl.tsx:27]: Only Base FormControl components don't need hard coded values, but Compound form controls like Pagination and Where clause can have such values. Please refer to WhereClauseControl file in https://github.com/appsmithorg/appsmith/pull/8490/files

**Review by rashmi-rai — APPROVED** [review]

**Review by ayushpahwa — APPROVED** [review]

**Pull request merged**